skeleton/src/main/java/org/killbill/commons/skeleton/metrics/TimedResourceInterceptor.java

```diff
@@ -29,33 +43,133 @@
  */
 public class TimedResourceInterceptor implements MethodInterceptor {
 
-    private final TimedResourceMetric timer;
+    private final Provider<GuiceContainer> jerseyContainer;
+    private final Provider<MetricRegistry> metricRegistry;
+    private final String resourcePath;
+    private final String metricName;
+    private String httpMethod;
 
-    public TimedResourceInterceptor(final TimedResourceMetric timer) {
-        this.timer = timer;
+    public TimedResourceInterceptor(Provider<GuiceContainer> jerseyContainer,
+                                    Provider<MetricRegistry> metricRegistry,
+                                    String resourcePath,
+                                    String metricName,
+                                    String httpMethod) {
+        this.jerseyContainer = jerseyContainer;
+        this.metricRegistry = metricRegistry;
+        this.resourcePath = resourcePath;
+        this.metricName = metricName;
+        this.httpMethod = httpMethod;
     }
 
     @Override
     public Object invoke(final MethodInvocation invocation) throws Throwable {
         final long startTime = System.nanoTime();
-        // two ways to get the response code:
-        // * response object is returned
-        // * exception is thrown
-        // * default response code (not sure how to get that ...)
-        Integer status = null;
-
+        int responseStatus = 0;
         try {
-            final Object result = invocation.proceed();
-
-            if (result instanceof Response) {
-                status = ((Response) result).getStatus();
+            final Object response = invocation.proceed();
+            if (response instanceof Response) {
+                responseStatus = ((Response) response).getStatus();
+            } else if (response == null || response instanceof Void) {
+                responseStatus = Response.Status.NO_CONTENT.getStatusCode();
+            } else {
+                responseStatus = Response.Status.OK.getStatusCode();
             }
```

Comment:
Yeah, a resource can return something like a String, or any object really. In that case Jersey returns a 200 response and the object as the body.